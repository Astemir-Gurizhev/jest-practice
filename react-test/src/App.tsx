import { useEffect, useState } from 'react'
import './App.css'

function App() {
  const [data, setData] = useState({})
  useEffect(()=> {
    setTimeout(()=> {
      setData({})
    },100)
  }, [])
	return <div className='App'>
    {data && <div style={{color:'red'}}>data</div>}
    <h1>hello world</h1>
    <button>click me</button>
    <input placeholder='input value' type="text" />
  </div>
}

export default App
