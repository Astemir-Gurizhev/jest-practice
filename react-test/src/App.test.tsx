import { render, screen } from '@testing-library/react'
import App from './App'

describe('react', () => {
	test('renders learn react link', () => {
		render(<App />)
		const helloWorldElem = screen.getByText(/hello world/i)
		const btn = screen.getByRole('button')
		const input = screen.getByPlaceholderText(/input value/i)
		expect(helloWorldElem).toBeInTheDocument()
		expect(btn).toBeInTheDocument()
		expect(input).toMatchSnapshot()
		
	})
	test('renders learn react link', async () => {
		render(<App/>)

		const helloWorldElem = await screen.findByText(/data/i)
		expect(helloWorldElem).toBeInTheDocument()
		expect(helloWorldElem).toHaveStyle({color:'red'})
	})
})