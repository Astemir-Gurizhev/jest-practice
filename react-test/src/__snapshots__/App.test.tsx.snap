// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapArrToStrings renders learn react link 1`] = `
<input
  placeholder="input value"
  type="text"
/>
`;

exports[`react renders learn react link 1`] = `
<input
  placeholder="input value"
  type="text"
/>
`;

exports[`renders learn react link 1`] = `
<input
  placeholder="input value"
  type="text"
/>
`;
